common/setup.go

```diff
@@ -78,8 +78,19 @@ func setupCluster(cfg *config.Config) (err error) {
 		}
 	} else {
 		log.Printf("CLUSTER_ID of '%s' was provided, skipping cluster creation and using it instead", cfg.ClusterID)
+
+		if cfg.ClusterName == "" {
+			cluster, err := OSD.GetCluster(cfg.ClusterID)
+			if err != nil {
+				return fmt.Errorf("could not retrieve cluster name: %v", err)
+			}
+
+			cfg.ClusterName = cluster.Name()
```

Comment:
Related to above, it might make sense to check for a zero-length string returned from `cluster.Name()` and return an error in that case.